connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -205,6 +205,91 @@ object ScalarUserDefinedFunction {
   }
 }
 
+case class UserDefinedAggregationFunction private[sql] (
```

Comment:
How much of this implementation is shared with the `ScalarUserDefinedFunction`?

Reply:
A lot. I just did a refactoring to re-use the ScalarUserDefinedFunction proto and most of the Scala code.